### PR TITLE
docs(examples): add simple example of v10 and v11 being used together

### PR DESCRIPTION
**Pull request by abbeyhrt**
REF #11986 
REF https://github.com/carbon-design-system/carbon/issues/12051

This PR adds a simple example of v10 and v11 working together. Most of the components are v10 but it uses the new Stack component and the v11 StructuredList. It's _super_ simple but I wanted to get something up quick that we could build on/add to 

#### Changelog

**New**

- {{new thing}}

**Changed**

- {{change thing}}

**Removed**

- {{removed thing}}

#### Testing / Reviewing

Pull down PR and `yarn` and `yarn dev` the `/incremental-migration` example, make sure it all looks okay.

### CODE REVIEW

File: examples/incremental-migration/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with
+[`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
+
+## Incremental migration
+
+We ave recently released a new version, v11, of our component library and a
+common question we receive is if users can adopt v11 incrementally and the
+answer is yes! If the burden of migrating to v11 is too big to take on all at
+once, but there are features of our latest release you want to take advantage
+of, it is likely that you can pull in just the parts yiu are wanting to use
```
Comment (tay1orjones) [inline examples/incremental-migration/README.md]:
```suggestion
of, it is likely that you can pull in just the parts you are wanting to use
```

File: examples/incremental-migration/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with
+[`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
+
+## Incremental migration
+
+We ave recently released a new version, v11, of our component library and a
```
Comment (jnm2377) [inline examples/incremental-migration/README.md]:
```suggestion
We have recently released a new version, v11, of our component library and a
```

File: examples/incremental-migration/README.md (hunk 1 above)
Comment (jnm2377) [inline examples/incremental-migration/README.md]:
```suggestion
of, it is likely that you can pull in just the parts you are wanting to use
```

**Review by jnm2377 — APPROVED** [review]
Looks great! Just commented some typo fixes :)

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**